api/notd/store/saver.py

```diff
@@ -477,6 +479,47 @@ async def update_latest_update(self, latestUpdateId: int, key: Optional[str] = N
         query = LatestUpdatesTable.update(LatestUpdatesTable.c.latestUpdateId == latestUpdateId).values(values)
         await self._execute(query=query, connection=connection)
 
+    async def create_token_attribute(self, registryAddress: str, tokenId: str, name: Optional[str], value: Optional[str], connection: Optional[DatabaseConnection] = None) -> TokenAttribute:
+        createdDate = date_util.datetime_from_now()
+        updatedDate = createdDate
+        values = {
+            TokenAttributesTable.c.createdDate.key: createdDate,
+            TokenAttributesTable.c.updatedDate.key: updatedDate,
+            TokenAttributesTable.c.registryAddress.key: registryAddress,
+            TokenAttributesTable.c.tokenId.key: tokenId,
+            TokenAttributesTable.c.name.key: name,
+            TokenAttributesTable.c.value.key: value,
+        }
+        query = TokenAttributesTable.insert().values(values)
+        result = await self._execute(query=query, connection=connection)
+        tokenAttributeId = result.inserted_primary_key[0]
+        return TokenAttribute(
+            tokenAttributeId=tokenAttributeId,
+            createdDate=createdDate,
+            updatedDate=updatedDate,
+            registryAddress=registryAddress,
+            tokenId=tokenId,
+            name=name,
+            value=value,
+        )
+
+    async def update_token_attribute(self, tokenAttributeId: int, name: Optional[str] = _EMPTY_STRING, value: Optional[str] = _EMPTY_STRING, connection: Optional[DatabaseConnection] = None) -> None:
+        values = {}
+        if name != _EMPTY_STRING:
+            values[TokenAttributesTable.c.name.key] = name
+        if value is not None:
```

Comment:
this is wrong - nullable fields should use the _EMPTY_STRING thing and non-nullable should not.